frontend/src/app/home/components/menu/search/search.component.ts

```diff
@@ -46,5 +54,42 @@ export class SearchComponent implements OnInit {
             });
         }
     }
+
+    makeSearchListEnabled() {
+        this.status = false;
+    }
+
+    makeSearchListDisabled() {
+        setTimeout(() => {if(document.activeElement != this.inputEl.nativeElement) this.status = true;}, 300)
+    }
+
+    autocomplete(text: string){
+        this.filteredOptions = this.mainService.options.filter(x => x.text.includes(text.toUpperCase()));
```

Comment:
zrobić consta z przefiltrowanymi i wykorzystac filtrowanie po `this.filteredOptions`

const options = this.filtredOptions.filter(x => x.text.includes(text.toUpperCase()));